plugins/modules/eos_pyateos.py

```diff
@@ -0,0 +1,568 @@
+        type: bool
+        default: false
+    group:
+        description:
+            - Pre set group of test. `group` and `test` are allowed togehter.
+            For more details: https://gitlab.com/networkAutomation/pyateos/-/blob/master/README.md
+        type: list
+        choices: [
+            'mgmt',
+            'routing',
+            'layer2',
+            'ctr',
+            'all'
+        ]
+    hostname:
+        description:
+            - Device hostname required for filesystem build
+        type: str
+        required: true
+requirements:
+    - jsondiff
+    - jmespath
+"""
+
+EXAMPLES = """
+---
+- name: run BEFORE tests.
+  eos_pyateos:
+    before: true
+    test:
+      - acl
+    group:
+      - mgmt
+      - layer2
+    hostname: "{{ inventory_hostname }}"
+    register: result
+
+- name: save BEFORE file IDs.
+  delegate_to: 127.0.0.1
+  set_fact:
+    before_ids: "{{ result.before_file_ids }}"
+
+- name: change mgmt config on switch.
+  eos_config:
+    lines:
+      - no ntp server vrf mgmt 10.75.33.5
+      - ntp server vrf mgmt 216.239.35.4
+
+- name: run AFTER tests.
+  eos_pyateos:
+    after: true
+    test:
+      - acl
+    group:
+      - mgmt
+      - layer2
+    hostname: "{{ inventory_hostname }}"
+    register: result
+
+- name: save AFTER file IDs.
+  delegate_to: 127.0.0.1
+  set_fact:
+    before_ids: "{{ result.after_file_ids }}"
+
+- name: run DIFF result.
+  eos_pyateos:
+    compare: true
+    group:
+      - mgmt
+      - layer2
+    test: "{{ tests }}"
+    hostname: "{{ inventory_hostname }}"
+    filter: true
+    files:
+      - "{{ before_ids }}"
+      - "{{ after_ids }}"
+"""
+
+RETURN = """
+compare:
+  description: before vs. after diff
+  returned: always
+  type: dict
+  sample: {
+    "delete": {
+        "ns2.sys.cloudsys.tmcs": {
+            "delay": 0.717,
+            "jitter": 0.117,
+            "lastReceived": 1583061550.0,
+            "peerType": "unicast",
+            "reachabilityHistory": [
+                true,
+                true
+            ],
+            "condition": "reject",
+            "offset": 0.047,
+            "peerIpAddr": "10.75.33.5",
+            "pollInterval": 64,
+            "refid": "169.254.0.1",
+            "stratumLevel": 3
+        }
+    },
+    "insert": {
+        "time2.google.com": {
+            "delay": 10.209,
+            "jitter": 0.0,
+            "lastReceived": 1583061584.0,
+            "peerType": "unicast",
+            "reachabilityHistory": [
+                true
+            ],
+            "condition": "reject",
+            "offset": -23.386,
+            "peerIpAddr": "216.239.35.4",
+            "pollInterval": 64,
+            "refid": ".GOOG.",
+            "stratumLevel": 1
+        }
+    }
+}
+"""
+import os
+import re
+import sys
+import json
+import time
+from ansible_collections.arista.eos.plugins.module_utils.basic import AnsibleModule, missing_required_lib
```

Comment:
`basic` is still part of ansible, and not this collection (:

```suggestion
from ansible.module_utils.basic import AnsibleModule, missing_required_lib
```